calculator.js

```diff
@@ -0,0 +1,47 @@
+const operators = ['+', '-', '*', '/'];
+
+exports.create_calculator = function (operator) {
+  const numbers = [];
+
+  return {
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
+        if (i === 0) {
+          i++;
+          return;
+        }
+
+        switch (operator) {
+          case '+':
+            answer = answer + number;
+            break;
+          case '-':
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
+        // console.log('current answer: ' + answer);
+
+        i++;
+      });
```

Comment:
using i to avoid adding the first number to itself here is a bit confusing. You could get rid of it and add another input parameter i to the function. foreach will autoincrement this for you similar to a for loop.

You could also consider using reduce without an initial value, instead of foreach